azkaban-common/src/main/java/azkaban/jobExecutor/ProcessJob.java

```diff
@@ -140,7 +140,7 @@ public void run() throws Exception {
         if (isMemGranted) {
           info(String.format("Memory granted for job %s", getId()));
           if (attempt > 1) {
-            CommonMetrics.INSTANCE.decrementOOMJobWaitCount();
+            SERVICE_PROVIDER.getInstance(CommonMetrics.class).decrementOOMJobWaitCount();
```

Comment:
Not sure if this is a good practice. If me, I would have a CommonMetrics's instance in the Constructor of `ProcessJob`. Basically, I don't know the performance/efficiency of using `SERVICE_PROVIDER.getInstance`